docs/requirements.txt

```diff
@@ -1,6 +1,7 @@
 breathe
 ipython
 numpydoc
-sphinx==2.4.4
+meson
 pydata-sphinx-theme
 sphinx-tabs
+sphinx>=4.2
```

Comment:
Can you maybe add a comment here in this file somewhere pointing to `conda_env_sphinx.txt` and with the message that those two files need to be kept in sync (so if changing the one, also need to update the other)?